contracts/contracts/layer_1/clearing/ClearingTransferFacet.sol

```diff
@@ -0,0 +1,455 @@
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+pragma solidity 0.8.18;
+
+// SPDX-License-Identifier: BSD-3-Clause-Attribution
+import {Common} from '../common/Common.sol';
+import {IClearingTransfer} from '../interfaces/clearing/IClearingTransfer.sol';
+import {IHold} from '../interfaces/hold/IHold.sol';
+import {_CLEARING_ROLE} from '../constants/roles.sol';
+import {
+    IStaticFunctionSelectors
+} from '../../interfaces/resolver/resolverProxy/IStaticFunctionSelectors.sol';
+import {_CLEARING_TRANSFER_RESOLVER_KEY} from '../constants/resolverKeys.sol';
+import {IKyc} from '../interfaces/kyc/IKyc.sol';
+
+// solhint-disable no-unused-vars, custom-errors
+contract ClearingTransferFacet is
+    IStaticFunctionSelectors,
+    IClearingTransfer,
+    Common
+{
+    function clearingTransferByPartition(
+        ClearingOperation calldata _clearingOperation,
+        uint256 _amount,
+        address _to
+    )
+        external
+        override
+        onlyUnpaused
+        onlyDefaultPartitionWithSinglePartition(_clearingOperation.partition)
+        onlyUnProtectedPartitionsOrWildCardRole
+        onlyWithValidExpirationTimestamp(_clearingOperation.expirationTimestamp)
+        validateAddress(_to)
+        onlyClearingActivated
+        returns (bool success_, uint256 clearingId_)
+    {
+        (success_, clearingId_) = _clearingTransferCreation(
+            _clearingOperation,
+            _amount,
+            _to,
+            _msgSender(),
```

Comment:
sender should be defined as a private variable to prevent all those calls.

Same in the rest of methods